src/Fabulous/Program.fs

```diff
@@ -94,12 +81,57 @@ type ProgramRunner<'arg, 'model, 'msg>(host: IHost, program: Program<'arg, 'mode
                 host.SetRootView(pageObj)
 
             lastViewDataOpt <- Some newPageElement
-                      
+    let viewInbox = MailboxProcessor.Start (fun inbox ->
+         let mutable isRendering = false
+         // We use local mutable instead of loop state here because we need a kind of "kill switch" for all events coming after render starts.
+         //If we use message for setting isRendered, this can lead to race conditions and two render processes may start simultaneously.
+         //Mutable variable helps us reset the blocking state immediately, so the next event occured would invoke render if needed.
+         let rec loop (state,hasChanges)  = async{
+             match! inbox.Receive() with
+             | ViewMsg.Render msg when (not isRendering)->
+                 isRendering<-true
+                 /// Here we put an inbox into the "waiting" state and wait for render finishes
+                 Debug.WriteLine "View invoked"
+                 async{
+                     program.syncAction (fun()->updateView msg) ()
+                     isRendering<-false
+                     inbox.Post (Completed)
+                 } |> Async.Start
+                 return! loop (state,false)
```

Comment:
I'm guessing `state` here should be `msg`, otherwise we will keep an old msg in the loop.

Reply:
Yes? tou are right